File: c++/Class_and_Object/Example-2/Student.cpp
#include <iostream>
using namespace std;

class Student {
public:
    string name;
    int rollNo;
    int marks1, marks2, marks3;

    Student(string n, int r, int m1, int m2, int m3) {
        name = n;
        rollNo = r;
        marks1 = m1;
        marks2 = m2;
        marks3 = m3;
    }

    void showDetails() {
        cout << "Name: " << name << " | RollNo: " << rollNo << endl;
        cout << "Marks: " << marks1 << ", " << marks2 << ", " << marks3 << endl;
        int total = marks1 + marks2 + marks3;
        double avg = total / 3.0;
        cout << "Total: " << total << " | Average: " << avg << endl;
    }
};

int main() {
    Student s1("SARAVANA", 101, 85, 90, 80);
    Student s2("DEVASRI", 102, 75, 88, 92);

    s1.showDetails();
    cout << endl;
    s2.showDetails();

    return 0;
}

File: Typescript/Class_and_Object/Example-2/Student.ts
class Student {
    constructor(
        public name: string,
        public rollNo: number,
        public marks1: number,
        public marks2: number,
        public marks3: number
    ) {}

    showDetails() {
        console.log(`Name: ${this.name} | RollNo: ${this.rollNo}`);
        console.log(`Marks: ${this.marks1}, ${this.marks2}, ${this.marks3}`);
        let total = this.marks1 + this.marks2 + this.marks3;
        let avg = total / 3;
        console.log(`Total: ${total} | Average: ${avg}`);
    }
}

let s1 = new Student("SARAVANA", 101, 85, 90, 80);
let s2 = new Student("DEVASRI", 102, 75, 88, 92);

s1.showDetails();
console.log();
s2.showDetails();
